tests/vmi_multus_test.go

```diff
@@ -101,39 +97,177 @@ var _ = Describe("Multus Networking", func() {
 			Expect(err).ToNot(HaveOccurred())
 			tests.WaitUntilVMIReadyWithNamespace("default", detachedVMI, tests.LoggedInCirrosExpecter)
 
-			cmdCheck := fmt.Sprintf("ping %s -c 1 -w 5\n", "10.1.1.1")
+			cmdCheck := "sudo /sbin/cirros-dhcpc up eth1 > /dev/null\n"
 			err = tests.CheckForTextExpecter(detachedVMI, []expect.Batcher{
 				&expect.BSnd{S: "\n"},
 				&expect.BExp{R: "\\$ "},
 				&expect.BSnd{S: cmdCheck},
 				&expect.BExp{R: "\\$ "},
-				&expect.BSnd{S: "echo $?\n"},
-				&expect.BExp{R: "0"},
-			}, 180)
+				&expect.BSnd{S: "ip addr show eth1 | grep 10.1.1 | wc -l"},
+				&expect.BExp{R: "1"},
+			}, 15)
 			Expect(err).ToNot(HaveOccurred())
 
 			By("checking virtual machine instance as two interfaces")
-			cmdCheck = fmt.Sprintf("ip link show %s\n", "eth0")
-			err = tests.CheckForTextExpecter(detachedVMI, []expect.Batcher{
-				&expect.BSnd{S: "\n"},
-				&expect.BExp{R: "\\$ "},
-				&expect.BSnd{S: cmdCheck},
-				&expect.BExp{R: "\\$ "},
-				&expect.BSnd{S: "echo $?\n"},
-				&expect.BExp{R: "0"},
-			}, 180)
+			checkInterface(detachedVMI, "eth0", "\\$ ")
+			checkInterface(detachedVMI, "eth1", "\\$ ")
+
+			pingVirtualMachine(detachedVMI, "10.1.1.1", "\\$ ")
+		})
+	})
+
+	Context("VirtualMachineInstance with ovs-cni plugin interface", func() {
+		nodeAffinity := &k8sv1.Affinity{
+			NodeAffinity: &k8sv1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
+					NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+						{
+							MatchExpressions: []k8sv1.NodeSelectorRequirement{
+								{Key: "kubernetes.io/hostname", Operator: k8sv1.NodeSelectorOpIn, Values: []string{nodes.Items[0].Name}},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		AfterEach(func() {
+			deleteVMIs(virtClient, []*v1.VirtualMachineInstance{vmiOne, vmiTwo})
+		})
+
+		It("should create two virtual machines with one interface", func() {
+			By("checking virtual machine instance can ping the secondary virtual machine instance using ovs-cni plugin")
+			// The virtual machines needs to run on the default namespace because the network-attachment-definitions.k8s.cni.cncf.io are there
+			vmiOne = tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskAlpine), "#!/bin/bash\n")
+			vmiOne.Namespace = "default"
```

Comment:
You can't run tests in the `default` namespace. You can use the constants `tests.NamespaceTestDefault` and `tests.NamespaceTestAlternative`. These two namespaces are managed by the test-suite and since there are two, also cross-namespace tests can be performed.

Reply:
@rmohr I know you have this namespaces but the network attachment crd are in the default namespace that why I need to create the vms there.

There is a way to apply a raw string containing the network attachment object from the test?
Then I can move the test into the regular test namespace.

Reply:
Ah, didn't know that they are namespaced. What you can do is something like this:

First fetch the object via a raw rest call, to work around the decoders, since they don't know the object:

```golang
virtClient.RestClient().Get().Resource("networkattachments").Namespace("default").Name("hhhhhhhhhhhhh").DoRaw()
```
then load the result into an `unstructed.Unstructed` struct:

```golang
func toUnstructured(raw []byte) *unstructured.Unstructured {
	var objmap map[string]interface{}
	err = json.Unmarshal(raw, &objmap)

	return &unstructured.Unstructured{Object: objmap}
}
```

The `unstructured` type allows manipulating the object in a convenient way. E.g. to exchange the namespace:

```go
unstructured.SetNestedField(obj, "newnamespace", "objectmeta", "namespace")
```

Finally upload it via a raw rest call:

```go
virtClient.RestClient().Put().Resource("networkattachments").Namespace("othernamespace").Name("hhhhhhhhhhhhh").Body(raw).DoRaw()
```

Reply:
@SchSeba what I described above would work for me as a temporary workaround. We should come up with a solution which does not depend on the default namespace in any way.

Reply:
@rmohr 
I tried to make the put as you suggest with this command `virtClient.RestClient().Put().Resource("NetworkAttachmentDefinition").Namespace(tests.NamespaceTestDefault).Body([]byte(ptpConfCRD)).Do()` 

But I get an error from the server `could not find the requested resource (put NetworkAttachmentDefinition.kubevirt.io)`

There is a way to remove the kubevirt.io?

Reply:
You need to use correct API collection and not KubeVirt one, I do not sure that we already have clientset that supports calls to `NetworkAttachmentDefinition`.

Reply:
The core client does not support `NetworkAttachmentDefinition` out of the box either. It's possible to simply use `RequestURI(...)` to do raw calls, instaed of `Resource` and `Namespace`. That should work.